NICE.Identity.Authorisation.WebAPI/Repositories/Context.cs

```diff
@@ -1,288 +1,322 @@
+                .Include(users => users.UserRoles)
+                    .ThenInclude(userRoles => userRoles.Role)
+                    .ThenInclude(website => website.Website)
+                .Include(email => email.UserEmailHistory)
+                    .ThenInclude(archived => archived.ArchivedByUser)
+                .ToList();
+
+            return !result.Any() ? null : result.Single();
         }
-        internal IEnumerable<User> GetUsersByOrganisationId(int organisationId)
-        {
-            return Jobs.Where(jobs => jobs.OrganisationId.Equals(organisationId))
-                           .Include(users => users)
-                            .Select(x => x.User)
-                           .ToList();
-        }
-
-        public List<User> FindUsers(string filter)
-        {
-	        filter ??= "";
-            
-	        var userIdsWithMatchingEmailHistory = 
-		        UserEmailHistory
-			        .Where(ueh => EF.Functions.Like(ueh.EmailAddress, $"%{filter}%"))
-			        .Select(ueh => ueh.UserId.Value)
-			        .ToList();
-
-	        return Users.Where(u => (u.FirstName != null && EF.Functions.Like(u.FirstName, $"%{filter}%"))
-	                                || (u.LastName != null && EF.Functions.Like(u.LastName, $"%{filter}%"))
-	                                || (u.FirstName != null && u.LastName != null && EF.Functions.Like(u.FirstName + " " + u.LastName, $"%{filter}%"))
-	                                || (u.NameIdentifier != null && EF.Functions.Like(u.NameIdentifier, $"%{filter}%"))
-	                                || (u.EmailAddress != null && EF.Functions.Like(u.EmailAddress, $"%{filter}%"))
-	                                || (userIdsWithMatchingEmailHistory.Contains(u.UserId)))
-
-		        .Include(users => users.UserRoles)
-					.ThenInclude(userRoles => userRoles.Role)
-						.ThenInclude(website => website.Website)
-
-					.OrderByDescending(user => user.UserId)
-		        .ToList();
+
+        public List<User> GetUsers(IEnumerable<string> authenticationProviderUserIds)
+        {
+            var lowercasedAuthenticationProviderUserIds = authenticationProviderUserIds.Select(userId => userId.ToLower());
+	        return Users.Where(users => lowercasedAuthenticationProviderUserIds.Contains(users.NameIdentifier.ToLower()))
+		        .Include(users => users.UserRoles)
+		        .ThenInclude(userRoles => userRoles.Role)
+		        .ThenInclude(website => website.Website)
+		        .ToList();
+        }
+        internal IEnumerable<User> GetUsersByOrganisationId(int organisationId)
+        {
+            return Jobs.Where(jobs => jobs.OrganisationId.Equals(organisationId))
+                           .Include(users => users)
+                            .Select(x => x.User)
+                           .ToList();
+        }
+
+        internal UserAndJobForOrganisation GetUsersAndJobsByOrganisationId(int organisationId)
+        {

```

Comment:
`UserAndJobForOrganisation`

Should this be plural? `UsersAndJobsForOrganisation`